docs/source/gateway.rst

```diff
@@ -0,0 +1,745 @@
+Each route section has a name, a type, and a model specification, which includes the model
+provider, name, and configuration details. The configuration section typically contains the base
+URL for the API and an environment variable for the API key.
+
+Here is an example of a simple single-route configuration:
+
+.. code-block:: yaml
+
+    routes:
+      - name: chat
+        type: llm/v1/chat
+        model:
+          provider: openai
+          name: gpt-3.5-turbo
+          config:
+            openai_api_base: https://api.openai.com/v1
+            openai_api_key: $OPENAI_API_KEY
+
+
+In this example, we define a route named ``chat`` that corresponds to the ``llm/v1/chat`` type, which
+will use the ``gpt-3.5-turbo`` model from `OpenAI` to return query responses from the `OpenAI` service.
+
+The beauty of the Gateway Configuration lies in its flexibility and ease of use.
+If the configuration file is updated at the target location, the `Mlflow Gateway` service can hot-swap
+routes, allowing new providers or model types to be incorporated without requiring changes to
+the applications interfacing with the gateway. The hot-swap functionality is autonomous and
+requires no server downtime or manual restarting of the service.
+This allows for simple, seamless changes to the underlying models and providers while keeping
+your applications steady and reliable.
+
+In order to define an API KEY for a given provider, there are three primary options:
+
+1. Directly include it in the YAML configuration file.
+2. Use an environment variable to store the API key and reference it in the YAML configuration file.
+3. Define your API key in a file and reference the location of that key-bearing file within the YAML configuration file.
+
+If you choose to include the API key directly, replace ``$OPENAI_API_KEY`` in the YAML file with your
+actual API key.
+
+If you prefer to use an environment variable (recommended), you can define it in your shell
+environment. For example:
+
+.. code-block:: bash
+
+     export OPENAI_API_KEY="your_openai_api_key"
+
+**Note:** Replace "your_openai_api_key" with your actual `OpenAI` API key.
+
+Remember to ensure secure access to the system that the `MLflow Gateway` service is running in to
+protect access to these keys.
+
+Routes
+------
+
+`Routes` are central to how the `MLflow Gateway` functions. Each route acts as a proxy endpoint for the
+user, channeling requests to the underlying model and provider specified in the configuration file.
+
+A route in the `MLflow Gateway` consists of the following fields:
+
+* **name**: This is the unique identifier for the route. This will be part of the URL when making API calls via the `MLflow Gateway`.
+
+* **type**: The type of the route corresponds to the type of language model interaction you desire. For instance, 'llm/v1/chat' for chat operations, 'llm/v1/completions' for text completion operations, and 'llm/v1/embeddings' for text embeddings.
```

Comment:
Can we call out completions and embeddings first, since they're more broadly supported? We only have Chat for OpenAI

```suggestion

* **type**: The type of the route corresponds to the type of language model interaction you desire. For instance, ``llm/v1/completions`` for text completion operations, ``llm/v1/embeddings`` for text embeddings, ``llm/v1/chat`` for chat operations.
```